src/Microsoft.ML.FastTree/GamModelParameters.cs

```diff
@@ -300,12 +300,25 @@ public double[] GetFeatureBinUpperBounds(int featureIndex)
             return featureBins;
         }
 
+        /// <summary>
+        /// Get all the bin upper bounds.
+        /// </summary>
+        public double[][] GetBinUpperBounds()
+        {
+            double[][] featureBins = new double[_binUpperBounds.Length][];
+            for (int i = 0; i < featureBins.Length; i++)
+            {
+                featureBins[i] = GetBinUpperBounds(i);
+            }
+            return featureBins;
+        }
+
         /// <summary>
         /// Get the binned weights for each feature.
```

Comment:
>weights [](start = 27, length = 7)

weights vs effects

Reply:
Done.

Reply:
It sill says weights

---
In reply to: [250005684](https://github.com/dotnet/machinelearning/pull/2188#discussion_r250005684) [](ancestors = 250005684)